File: src/main/java/ar/edu/um/tesoreria/rest/extern/model/PreInscripcionFacultad.java
/**
 * 
 */
package ar.edu.um.tesoreria.rest.extern.model;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.OffsetDateTime;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @author daniel
 *
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class PreInscripcionFacultad implements Serializable {
	/**
	 * 
	 */
	private static final long serialVersionUID = -5119519803283924663L;

	private Long preinscripcionId;
	private Integer facultadId;
	private Integer lectivoId;
	private Integer geograficaId;
	private Integer turnoId;
	private BigDecimal personaId;
	private Integer documentoId;
	private OffsetDateTime fecha;
	private String chequera;
	private String observaciones;
	private String barras;

	public String getPersonaKey() {
		return this.personaId + "." + this.documentoId;
	}

}

File: src/main/java/um/tesoreria/core/extern/model/kotlin/PreInscripcionFacultad.kt
package um.tesoreria.core.extern.model.kotlin

import java.math.BigDecimal
import java.time.OffsetDateTime

data class PreInscripcionFacultad(

    var preinscripcionId: Long? = null,
    var facultadId: Int? = null,
    var lectivoId: Int? = null,
    var geograficaId: Int? = null,
    var turnoId: Int? = null,
    var personaId: BigDecimal? = null,
    var documentoId: Int? = null,
    var fecha: OffsetDateTime? = null,
    var chequera: String? = null,
    var observaciones: String? = null,
    var barras: String? = null,

    ) {
    fun getPersonaKey(): String? {
        return personaId.toString() + "." + documentoId
    }
}
